packages/@aws-cdk/cdk/lib/context.ts

```diff
@@ -17,38 +19,48 @@ export class ContextProvider {
 
   private readonly stack: Stack;
 
-  constructor(private context: Construct) {
+  constructor(
+    private readonly context: Construct,
+    private readonly provider: string,
+    private readonly props: {[key: string]: any} = {}) {
     this.stack = Stack.find(context);
   }
 
+  public get key(): string {
+    const propStrings: string[] = propsToArray({
+      ...this.props,
+      ...{account: this.account, region: this.region},
+    });
+    return `${this.provider}:${propStrings.join(':')}`;
+  }
   /**
    * Read a provider value, verifying it's a string
    * @param provider The name of the context provider
    * @param scope The scope (e.g. account/region) for the value
    * @param args Any arguments
    * @param defaultValue The value to return if there is no value defined for this context key
    */
-  public getStringValue(
-    provider: string,
-    scope: undefined | string[],
-    args: string[],
-    defaultValue: string): string {
+  public getStringValue( defaultValue: string): string {
     // if scope is undefined, this is probably a test mode, so we just
     // return the default value
-    if (!scope) {
-      this.context.addError(formatMissingScopeError(provider, args));
+    if (!this.account || !this.region) {
```

Comment:
Interestingly, this now means something else. Because previously there was a distinction between "not having a scope" and "not having an account/region", and now there isn't anymore.

This only comes into play if we have context providers that don't need both an account and a region, which we don't have right now anyway so I'm okay with this change and we can always fix it once we do. But it's something to consider.